platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/statuslogic/PlatformStatusLogic.java

```diff
@@ -0,0 +1,137 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.system.platformstatus.statuslogic;
+
+import com.swirlds.common.system.platformstatus.PlatformStatus;
+import com.swirlds.common.system.platformstatus.statusactions.CatastrophicFailureAction;
+import com.swirlds.common.system.platformstatus.statusactions.DoneReplayingEventsAction;
+import com.swirlds.common.system.platformstatus.statusactions.FallenBehindAction;
+import com.swirlds.common.system.platformstatus.statusactions.FreezePeriodEnteredAction;
+import com.swirlds.common.system.platformstatus.statusactions.PlatformStatusAction;
+import com.swirlds.common.system.platformstatus.statusactions.ReconnectCompleteAction;
+import com.swirlds.common.system.platformstatus.statusactions.SelfEventReachedConsensusAction;
+import com.swirlds.common.system.platformstatus.statusactions.StartedReplayingEventsAction;
+import com.swirlds.common.system.platformstatus.statusactions.StateWrittenToDiskAction;
+import com.swirlds.common.system.platformstatus.statusactions.TimeElapsedAction;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Interface representing the state machine logic for an individual {@link PlatformStatus}.
+ * <p>
+ * The methods in this interface that process {@link PlatformStatusAction}s behave in the following way:
+ * <ul>
+ *     <li>If the input action results in a status transition, the processing method should return an instance of
+ *     {@link PlatformStatusLogic} corresponding to the new status</li>
+ *     <li>If the input action does not result in a status transition, the processing method should return a reference
+ *     to itself, since it will continue managing the logic for the current status status moving forward</li>
+ *     <li>If the input action is not a valid for the current status, the processing method should throw an
+ *     {@link com.swirlds.common.system.platformstatus.IllegalPlatformStatusException IllegalPlatformStatusException}</li>
+ * </ul>
+ */
+public interface PlatformStatusLogic {
+    /**
+     * Process a {@link CatastrophicFailureAction}
+     *
+     * @param action the action to process
+     * @return the {@link PlatformStatusLogic} to manage the resulting status
+     */
+    @NonNull
+    PlatformStatusLogic processCatastrophicFailureAction(@NonNull final CatastrophicFailureAction action);
+
+    /**
+     * Process a {@link DoneReplayingEventsAction}
+     *
+     * @param action the action to process
+     * @return the {@link PlatformStatusLogic} to manage the resulting status
+     */
+    @NonNull
+    PlatformStatusLogic processDoneReplayingEventsAction(@NonNull final DoneReplayingEventsAction action);
```

Comment:
I ask myself if all the method should have the following default implementation:

```
Objects.requireNonNull(action);
throw new IllegalPlatformStatusException(action, getStatus());
```

Reply:
We discussed this [above](https://github.com/hashgraph/hedera-services/pull/7060#discussion_r1231223328)

TLDR; I think it's worth forcing anyone touching this code to carefully consider the effect of an action on every individual status, without relying on default implementations.